include/aspect/particle/property/interface.h

```diff
@@ -0,0 +1,355 @@
+           * does nothing, so that property classes that do not take any
+           * parameters do not have to do anything at all.
+           *
+           * This function is static (and needs to be static in derived
+           * classes) so that it can be called without creating actual objects
+           * (because declaring parameters happens before we read the input
+           * file and thus at a time when we don't even know yet which
+           * property objects we need).
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters this class declares from the parameter file.
+           * The default implementation in this class does nothing, so that
+           * derived classes that do not need any parameters do not need to
+           * implement it.
+           */
+          virtual
+          void
+          parse_parameters (ParameterHandler &prm);
+      };
+
+
+
+      /**
+       * Manager class of properties - This class sets the data of the particles
+       * and updates it over time if requested by the user selected properties
+       */
+      template <int dim>
+      class Manager : public SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * Empty constructor for Manager
+           */
+          Manager ();
+
+          /**
+           * Destructor for Manager
+           */
+          virtual
+          ~Manager ();
+
+          /**
+           * Initialization function. This function is called once at the
+           * beginning of the program after parse_parameters is run.
+           */
+          virtual
+          void
+          initialize ();
+
+          /**
+           * Initialization function for particle properties. This function is
+           * called once at the creation of a particle
+           */
+          virtual
+          void
+          initialize_particle (Particle<dim> &particle,
+                               const Vector<double> &solution,
+                               const std::vector<Tensor<1,dim> > &gradients);
+
+          /**
+           * Update function for particle properties. This function is
+           * called once every timestep for every particle
+           */
+          virtual
+          void
+          update_particle (Particle<dim> &particle,
+                           const Vector<double> &solution,
+                           const std::vector<Tensor<1,dim> > &gradients);
+
+          /**
+           * Returns a bool, which is false if no selected plugin needs to
+           * update tracer properties over time. This saves considerable
+           * computation time in cases, when no plugin needs to update tracer
+           * properties over time, because the solution does not need to be
+           * evaluated at tracer positions in this case.
+           */
+          virtual
+          UpdateTimeFlags
+          need_update ();
```

Comment:
Does this function (in the manager) need to be virtual? Do you expect classes to be derived from the manager class?